vignettes/namespace.Rmd

```diff
@@ -19,53 +19,118 @@ This is a little frustrating at first, but soon becomes second-nature.
+```{r}
+#' Add two numbers together
+#' 
+#' @param x,y A pair of numbers.
+#' @export
+add <- function(x, y) {
+  x + y
+}
+```
 
--   **S3 methods**: every S3 method *must* be exported, even if the generic is not.
-    Otherwise, the S3 method table will not be generated correctly and internal generics will not find the correct method.
-    See below for instructions on how to export a method for a generic in a suggested package.
+### S3
 
--   **S4 classes**: export S4 classes if you want others to be able to extend them.
+An S3 generic works like a regular R function so export it following the advice above: if you want users to call it, export; otherwise, don't.
 
--   **S4 generics:** `@export` if you want the generic to be publicly usable.
+```{r}
+#' Take an object to bizarro world
+#' 
+#' @param x A vector.
+#' @export
+bizarro <- function(x, ...) {
+  UseMethod("bizarro")
+}
+```
 
--   **S4 methods**: you only need to `@export` methods for generics in other packages.
+While S3 methods are regular functions with a special naming scheme, their "export" works a bit differently.
+S3 methods are exported only in the sense that calling the generic with the appropriate class will call the method; a user can't directly access the method definition by typing its name.
+A more technically correctly term would be to say that the method is **registered** so that the generics can find it.
 
-### S3 methods for generics in suggested packages
+You must register, i.e. `@export`, every S3 method regardless of whether or not the generic is exported.
+roxygen2 will warn you if you have forgotten.
 
-`@exportS3Method` tag allows you to generate `S3method()` namespace directives.
-Its primary use is for "delayed" method registration, which allows you to define methods for generics found in suggested packages (R \>= 3.6).
-For example,
+```{r}
+#' @export
+bizarro.character <- function(x, ...) {
+  letters <- strsplit(x, "")
+  letters_rev <- lapply(letters, rev)
+  vapply(letters_rev, paste, collapse = "", FUN.VALUE = character(1))
+}
+```
+
+You have four options for documenting an S3 method:
+
+-   Don't document it; it's not required, and not needed for simple generics where the user won't care about the details.
+-   If the method is particularly complex or has many arguments that the generic does not, you can document it in its own file. In this case, just document it as if its a function.
+-   You can use `@rdname` to document it with other methds for the generic. This is a good option if it's your generic, and you're providing a bunch of methods for different classes.
+-   You can use `@rdname` to document it with other methods for the class. This is typically the least appealing option because the different generics will have different arguments, leading to a cluttered and potentially confusing page.
+
+```{r}
+#' Take an object to bizarro world
+#' 
+#' @description
```

Comment:
We don't normally use `@description` explicitly. Is there a reason to do so in the current context?

Reply:
Because I think you want the bulleted listed of methods above the fold?  (i.e. we do use an explicit `@description` when there's a list)